docs/docs/deploying.md

```diff
@@ -24,26 +24,69 @@ If you're using [CronJobs()](/api/cronjob), make sure to run `quirrel ci` during
+## How to deploy to your own server
 
-Here's a rough guideline on how to acquire tokens:
+If you still want to host Quirrel yourself, there are currently two ways to host our own quirrel server:
 
-1. Think of a good password and supply it via the `PASSPHRASES` env var.
-2. Start up your instance and run `curl --user ignored:put_password_here -X PUT localhost:9181/tokens/local`
-3. Use the obtained token for setting your application's `QUIRREL_TOKEN` env var.
+1. [By using the Docker Image](#using-the-docker-image).
+2. [By launching our own instance](#launching-our-own-instance).
 
-Using self-hosted Quirrel requires you to set the `QUIRREL_API_URL` variable to the location of your deployment (it defaults to `https://api.quirrel.dev`).
-The `QUIRREL_TOKEN` can be obtained using the server's [REST API](https://api.quirrel.dev/documentation/index.html#/default/put_tokens__id_). If a passphrase was set, it must be passed in using basic authentication.
+Both of them require that our connections are secured through a secured token.
 
+### Deploying the server
+
+**Please note that the deployment should be secured using HTTPS.**
+
+#### Using the Docker Image
+
+Find Quirrel's docker image here: https://github.com/orgs/quirrel-dev/packages/container/package/quirrel
+
+To use Quirrel's Docker Image, we need to set the connection string at
+`REDIS_URL`. **Make sure your Redis instance is persistent**.
+
+Also, we'll need to set a `PASSPHRASE` (or more, separated by colons `:`) that will be used in the next step to generate a token. You can do so by setting the environment variable:
+
+```
+PASSPHRASES=PASSPHRASE1[:PASSPHRASE2[:PASSPHRASE3]]
+```
+
+#### Launching our own instance
+
+First, we will fire up our server with a `PASSPHRASE` (or more, separated by colons `:`) that will be used in the next step to generate a token. To do so, launch quirrel like so:
+
+```
+quirrel --passphrase PASSPHRASE1[:PASSPHRASE2[:PASSPHRASE3]]
+```
+
+### Acquiring a token
+
+Once our server is ready, we'll need to [get an auth token](https://api.quirrel.dev/documentation/static/index.html#/Admin/put_tokens__id_):
+
+`curl --user ignored:{PASSPHRASE} -X PUT {QUIRREL_SERVER_URL}/tokens/{NAME_OF_TOKEN}`
+
+This petition will return a token. Save it for the next and last step.
+
+### Configuring our client to auth connections with our token
+
+Lastly, we should configure our client to use the newly generated **token** as well as provide quirrel with our own client's base url. To do so, add the following environment variables to your client:
+
+```
+# example ./.env file in the client's root directory
+
+# Client's base url
+QUIRREL_API_URL=http://localhost:3000
```

Comment:
```suggestion
QUIRREL_API_URL=http://your-quirrel-api-address:9181
```

`QUIRREL_API_URL` denotes where the API is reachable. I don't think we need to mention `QUIRREL_BASE_URL` here, as it's not specific to self-hosted deployments.